SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -876,6 +880,66 @@ void FSpatialGDKEditorToolbarModule::LaunchInspectorWebpageButtonClicked()
 	}
 }
 
+void FSpatialGDKEditorToolbarModule::LaunchInspectorWebpageButtonClicked()
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	const FString InspectorVersion = SpatialGDKEditorSettings->GetInspectorVersion();
+
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, InspectorVersion] {
+		if (InspectorProcess && InspectorProcess->Update())
+		{
+			// We already have an inspector running. Just open the URL.
+			OpenInspectorURL();
+			return;
+		}
+
+		// Check for any old inspector processes that may be leftover from previous runs. Kill any we find.
+		SpatialCommandUtils::TryKillProcessWithName(SpatialGDKServicesConstants::InspectorExe);
+
+		// Grab the inspector binary
+		if (!SpatialCommandUtils::FetchInspectorBinary(InspectorVersion))
+		{
+			UE_LOG(LogSpatialGDKEditorToolbar, Error, TEXT("Attempted to fetch the local inspector binary but failed!"));
+			OnShowFailedNotification(TEXT("Failed to fetch local inspector!"));
+		}
+
+		FString InspectorArgs =
+			FString::Printf(TEXT("--backend_addr=%s --schema_bundle=\"%s\""), *SpatialGDKServicesConstants::InspectorIPPort,
+							*SpatialGDKServicesConstants::SchemaBundlePath);
+
+		InspectorProcess = { *SpatialGDKServicesConstants::GetInspectorExecutablePath(InspectorVersion), *InspectorArgs,
+							 SpatialGDKServicesConstants::SpatialOSDirectory, /*InHidden*/ true,
+							 /*InCreatePipes*/ true };
+
+		FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+		TWeakPtr<SSpatialOutputLog> SpatialOutputLog = GDKServices.GetSpatialOutputLog();
+
+		bool bStartingInspector = true;
+
+		InspectorProcess->OnOutput().BindLambda([this, &bStartingInspector](const FString& Output) {
+			if (Output.Contains(FString::Printf(TEXT("Open %s"), *SpatialGDKServicesConstants::InspectorV2URL)))
+			{
+				bStartingInspector = false;
+			}
+
+			UE_LOG(LogSpatialGDKEditorToolbar, Log, TEXT("Inspector: %s"), *Output)
+		});
+
+		InspectorProcess->Launch();
+
+		while (bStartingInspector && InspectorProcess->Update())
+		{
+			if (InspectorProcess->GetDuration().GetTotalSeconds() > 5)
```

Comment:
think we discussed this before (different PR, but same pattern :P ) - if we really need to have our own time out, let's make it configurable. though if possible I would still prefer to just have the time out passed in the command line args for the inspector process, if a timeout flag for it exists. And then we can just have bind to OnCanceled to print the error.

Reply:
Removed the timeout all together as realised it's not needed for web page opening, if the page is opened before the inspector is ready then it will simply refresh when the port opens (to my understanding).

Good idea on binding to the OnCancelled though. Will do that with a notification. There is no timeout flag for the inspector binary, it was just an attempt at some helper code to make the UX a bit nicer.